### PR TITLE
[improvement] Added traceID in all cases

### PR DESCRIPTION
**Pull request by buffer51**
## Before this PR
If an HTTPError's response was not valid JSON, only the error message and response text would be shown.

## After this PR
Now, the error message, response text and traceID (if available, otherwise defaults to `None`) are shown.

This adds on top of #25 to improve experience when debugging errors.
Replaces #34.

### CODE REVIEW

**Review by ahggns — APPROVED** [review]

File: conjure_python_client/_http/requests_client.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def _request(self, *args, **kwargs):
         try:
             _response.raise_for_status()
         except HTTPError as e:
-            if e.response is not None and e.response.content is not None:
+            if e.response is not None:
```
Comment (ferozco) [inline conjure_python_client/_http/requests_client.py:80]: why was this changed?

Comment (buffer51) [reply conjure_python_client/_http/requests_client.py:80]: We want to handle a response with no content with the ConjureHTTPError so we can show the HTTP message & trace ID. Without that it will not reach the block I modified to add the trace ID.

**Review by ferozco — APPROVED** [review]
